ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListQueriesExecutor.java

```diff
@@ -40,25 +50,93 @@ private ListQueriesExecutor() { }
-          statement.getStatementText(),
-          executionContext.getPersistentQueries().stream()
-              .map(QueryDescriptionFactory::forQueryMetadata)
-              .collect(Collectors.toList())));
+      return executeExtended(
+          listQueries,
+          statement,
+          sessionProperties,
+          executionContext,
+          serviceContext);
+    }
+
+    final List<RunningQuery> runningQueries = new ArrayList<>();
+    runningQueries.addAll(executionContext.getPersistentQueries()
+        .stream()
+        .map(q -> new RunningQuery(
+            q.getStatementString(),
+            ImmutableSet.of(q.getSinkName().name()),
+            ImmutableSet.of(q.getResultTopic().getKafkaTopicName()),
+            q.getQueryId(),
+            Optional.of(q.getState()),
+            Optional.of(new KsqlHostInfoEntity(sessionProperties.getKsqlHostInfo()))
+        ))
+        .collect(Collectors.toList()));
+    
+    if (!sessionProperties.getIsInternalRequest()) {
+      final Set<HostInfo> hosts =
+          DiscoverRemoteHostsUtil.getRemoteHosts(
+              executionContext.getPersistentQueries(),
+              sessionProperties.getKsqlHostInfo());
+
+      hosts.forEach(hostInfo -> {
+        final KsqlEntityList response = serviceContext.getKsqlClient().makeInternalKsqlRequest(
+            ServerUtil.buildRemoteUri(
+                sessionProperties.getLocalUrl(),
+                hostInfo.host(),
+                hostInfo.port()
+            ),
+            "show queries;"
+        ).getResponse();
+        response.forEach(queries -> {
+          runningQueries.addAll(((Queries) queries).getQueries());
+        });
+      });
     }
 
     return Optional.of(new io.confluent.ksql.rest.entity.Queries(
         statement.getStatementText(),
-        executionContext.getPersistentQueries()
-            .stream()
-            .map(q -> new RunningQuery(
-                q.getStatementString(),
-                ImmutableSet.of(q.getSinkName().name()),
-                ImmutableSet.of(q.getResultTopic().getKafkaTopicName()),
-                q.getQueryId(),
-                Optional.of(q.getState())
-            ))
-            .collect(Collectors.toList())));
+        runningQueries));
   }
 
+  private static Optional<KsqlEntity> executeExtended(
+      final ListQueries listQueries,
+      final ConfiguredStatement<ListQueries> statement,
+      final SessionProperties sessionProperties,
+      final KsqlExecutionContext executionContext,
+      final ServiceContext serviceContext
+  ) {
+    final List<QueryDescription> queryDescriptionList = new ArrayList<>();
+    queryDescriptionList.addAll(
+        executionContext.getPersistentQueries().stream()
+        .map(query ->
+            QueryDescriptionFactory.forQueryMetadata(
+                query,
+                Optional.of(new KsqlHostInfoEntity(sessionProperties.getKsqlHostInfo()))
+            ))
+        .collect(Collectors.toList()));
```

Comment:
Rather than building a list containing all the persistent queries and then adding all elements in that list to another list, just work with streams:

```java
    executionContext.getPersistentQueries().stream()
        .map(query ->
            QueryDescriptionFactory.forQueryMetadata(
                query,
                Optional.of(new KsqlHostInfoEntity(sessionProperties.getKsqlHostInfo()))
            ))
    .forEach(queryDescriptionLists::add);
```

Is there anywhere else in this PR where this same pattern can be applied?